storage/tree_id.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"crypto/rand"
+	"math"
+	"math/big"
+)
+
+// NewTreeID generates a random, positive, non-zero tree ID.
+func NewTreeID() (int64, error) {
+	id, err := rand.Int(rand.Reader, big.NewInt(math.MaxInt64))
```

Comment:
Discussed offline.

TL;DR: I don't want to use math/rand, pick some seed and risk IDs being predictable. It doesn't seem like it would be a problem anyway, but I think we get a better mileage from crypto/rand regardless.